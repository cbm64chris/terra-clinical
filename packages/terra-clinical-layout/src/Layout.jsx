import React from 'react';
import PropTypes from 'prop-types';
import AppDelegate from 'terra-app-delegate';
import ContentContainer from 'terra-content-container';

import LayoutSlidePanel from './_LayoutSlidePanel';
import MenuHeader from './_MenuHeader';
import {
  getBreakpointSize,
  getCustomProps,
} from './utilities';

const propTypes = {
  /**
   * The AppDelegate instance provided by the containing component.
   */
  app: AppDelegate.propType,
  /**
   * ELement to be placed within the toolbar section of the layout.
   */
  toolbar: PropTypes.element,
  /**
   * Element to be placed within the main content section of the layout.
   */
  content: PropTypes.element,
  /**
   * Element to be placed within the menu section of the layout.
   */
  menu: PropTypes.element,
  /**
   * Whether or not menu toggling is enabled. If not enabled, menu content will not be visible, and toggle
   * functionality will be hidden.
   */
  enableMenu: PropTypes.bool,
};

const defaultProps = {
  enableMenu: false,
};

const compactSizes = ['tiny', 'small'];
const isSizeCompact = size => (
  compactSizes.indexOf(size) >= 0
);

class Layout extends React.Component {
  constructor(props) {
    super(props);

    this.toggleMenu = this.toggleMenu.bind(this);
    this.togglePin = this.togglePin.bind(this);
    this.updateSize = this.updateSize.bind(this);
    this.isCompactLayout = this.isCompactLayout.bind(this);
    this.renderToolbar = this.renderToolbar.bind(this);
    this.renderMenu = this.renderMenu.bind(this);
    this.renderContent = this.renderContent.bind(this);

    this.state = {
      menuIsOpen: false,
      menuIsPinned: false,
      menuIsEnabled: props.enableMenu,
      size: getBreakpointSize(),
    };
  }

  componentDidMount() {
    window.addEventListener('resize', this.updateSize);
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.enableMenu !== this.props.enableMenu) {
      this.setState({
        menuIsEnabled: nextProps.enableMenu,
        menuIsOpen: nextProps.enableMenu && this.state.menuIsOpen,
        menuIsPinned: nextProps.enableMenu && this.state.menuIsPinned,
      });
    }
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.updateSize);
  }

  updateSize() {
    const newSize = getBreakpointSize();

    if (this.state.size !== newSize) {
      const newMenuIsPinned = this.props.enableMenu && !isSizeCompact(newSize) && this.state.menuIsPinned;
      const newMenuIsOpen = this.props.enableMenu && this.state.menuIsOpen && newMenuIsPinned;

      this.setState({
        size: newSize,
        menuIsOpen: newMenuIsOpen,
        menuIsPinned: newMenuIsPinned,
      });
    }
  }

  toggleMenu() {
    if (!this.state.menuIsPinned) {
      this.setState({
        menuIsOpen: !this.state.menuIsOpen,
      });
    }
  }

  togglePin() {
    this.setState({
      menuIsPinned: !this.state.menuIsPinned,
    });
  }

  isCompactLayout() {
    return isSizeCompact(this.state.size);
  }

  renderToolbar() {
    const { app, toolbar } = this.props;
    const { size, menuIsOpen, menuIsEnabled } = this.state;

    if (!toolbar) {
      return null;
    }

    const isCompactLayout = this.isCompactLayout();
    const shouldDisplayMenuToggle = isCompactLayout && menuIsEnabled;

    return React.cloneElement(toolbar, {
      app,
      layoutConfig: {
        size,
        isCompactLayout,
        toggleMenu: shouldDisplayMenuToggle && this.toggleMenu,
        menuIsOpen,
      },
    });
  }

  renderMenu() {
    const { app, menu } = this.props;
    const { size, menuIsOpen, menuIsPinned } = this.state;

    if (!menu) {
      return null;
    }

    const isCompactLayout = this.isCompactLayout();
    const vessel = React.cloneElement(menu, {
      app,
      layoutConfig: {
        size,
        isCompactLayout,
        toggleMenu: this.toggleMenu,
        menuIsOpen,
      },
    });

    return (
      <ContentContainer
        fill
        header={!isCompactLayout ? (
          <MenuHeader
            text="Menu"
            togglePin={!isCompactLayout ? this.togglePin : undefined}
            isPinned={!isCompactLayout ? menuIsPinned : undefined}
          />
        ) : undefined}
      >
        {vessel}
      </ContentContainer>
    );
  }

  renderContent() {
    const { app, content } = this.props;
    const { size, menuIsOpen } = this.state;

    if (!content) {
      return null;
    }

    const isCompactLayout = this.isCompactLayout();

    return (
      <ContentContainer
        fill
        header={isCompactLayout && this.renderToolbar()}
      >
        {(
          React.cloneElement(content, {
            app,
            layoutConfig: {
              size,
              isCompactLayout,
              toggleMenu: this.toggleMenu,
              menuIsOpen,
            },
          })
        )}
      </ContentContainer>
    );
  }

  render() {
    const { menuIsOpen, menuIsPinned, menuIsEnabled, size } = this.state;

    return (
      <ContentContainer
        fill
        header={!this.isCompactLayout() && this.renderToolbar()}
        {...getCustomProps(this.props, propTypes)}
      >
        <LayoutSlidePanel
          isAnimated
          isOpen={menuIsOpen}
          onRequestClose={this.toggleMenu}
          panelBehavior={menuIsPinned ? 'squish' : 'overlay'}
          panelContent={this.renderMenu()}
          mainContent={this.renderContent()}
          size={size}
          toggleMenu={this.toggleMenu}
          isToggleEnabled={menuIsEnabled}
        />
      </ContentContainer>
    );
  }
}

Layout.propTypes = propTypes;
Layout.defaultProps = defaultProps;

export default Layout;
export { getBreakpointSize, getCustomProps };
