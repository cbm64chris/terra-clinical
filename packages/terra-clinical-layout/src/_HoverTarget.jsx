import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import IconChevronRight from 'terra-icon/lib/icon/IconChevronRight';
import 'terra-base/lib/baseStyles';

import styles from './HoverTarget.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * The component to display in the panel content area.
   */
  children: PropTypes.node,
  /**
   * Flag to enable the hover functionality.
   */
  isHoverEnabled: PropTypes.bool,
  /**
   * Flag to menu .
   */
  isOpen: PropTypes.bool,
  /**
   * Function callback to trigger when the hover section is clicked on mobile.
   */
  onClick: PropTypes.func,
  /**
   * Function callback to trigger when the mouse leaves.
   */
  onHoverOff: PropTypes.func,
  /**
   * Function callback to trigger when the mouse enters.
   */
  onHoverOn: PropTypes.func,
  /**
   * String text to display.
   */
  text: PropTypes.string,
};

const defaultProps = {
  children: [],
  isHoverEnabled: false,
  isEnabled: false,
};

class HoverTarget extends React.Component {
  constructor(props) {
    super(props);
    this.setHoverNode = this.setHoverNode.bind(this);
    this.handleMouseEnter = this.handleMouseEnter.bind(this);
    this.handleMouseLeave = this.handleMouseLeave.bind(this);
    this.handleOnClick = this.handleOnClick.bind(this);
    this.updateSize = this.updateSize.bind(this);

    this.listenersAdded = false;
    this.isMouseEnterActive = false;
  }

  componentDidMount() {
    window.addEventListener('resize', this.updateSize);

    this.updateSize();

    if (this.hoverNode) {
      this.listenersAdded = this.updateListenersOnNode(this.hoverNode);
    }
  }

  componentDidUpdate() {
    if (this.hoverNode) {
      this.listenersAdded = this.updateListenersOnNode(this.hoverNode);
    }
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.updateSize);

    if (this.hoverNode) {
      this.listenersAdded = this.removeListenersFromNode(this.hoverNode);
    }
  }

  setHoverNode(node) {
    this.hoverNode = node;
  }

  updateListenersOnNode(node) {
    if (this.props.isHoverEnabled) {
      return this.addListenersToNode(node);
    } else if (!this.props.isHoverEnabled) {
      return this.removeListenersFromNode(node);
    }
    return false;
  }

  updateSize() {
    // Due to the nature of the rotated text and therefore inverted coordinates, we need to update
    // the width of the text element to match the available height of the container.
    this.textElement.style.width = `${this.textContainerElement.offsetHeight}px`;
  }

  addListenersToNode(node) {
    if (!this.listenersAdded) {
      node.addEventListener('mouseenter', this.handleMouseEnter);
      node.addEventListener('mouseleave', this.handleMouseLeave);
      this.isMouseEnterActive = false;
    }
    return true;
  }

  removeListenersFromNode(node) {
    if (this.listenersAdded) {
      node.removeEventListener('mouseenter', this.handleMouseEnter);
      node.removeEventListener('mouseleave', this.handleMouseLeave);
      this.isMouseEnterActive = false;
    }
    return false;
  }

  handleMouseEnter(event) {
    if (this.props.onHoverOn && !this.isMouseEnterActive) {
      this.props.onHoverOn(event);
    }
    this.isMouseEnterActive = true;
  }

  handleMouseLeave(event) {
    if (this.props.onHoverOff) {
      this.props.onHoverOff(event);
    }
    this.isMouseEnterActive = false;
  }

  handleOnClick(event) {
    if (this.props.onClick) {
      this.props.onClick(event);
    }
  }

  render() {
    const {
      children,
      isHoverEnabled,
      isOpen,
      onClick,
      onHoverOff,
      onHoverOn,
      text,
      ...customProps
    } = this.props;

    const hoverClasses = cx([
      'hover-menu',
      { 'is-hover-disabled': !isHoverEnabled },
      { 'is-hover-tab-hidden': isOpen },
      customProps.className,
    ]);

    return (
      <div {...customProps} className={hoverClasses} ref={this.setHoverNode}>
        <div className={cx('content-section')}>
          {children}
        </div>
        <button className={cx('hover-section')} onClick={this.handleOnClick}>
          <div className={cx('hover-section-icon')}>
            <IconChevronRight className={cx('hover-icon')} />
          </div>
          <div ref={(element) => { this.textContainerElement = element; }} className={cx('hover-section-text')}>
            <h3 ref={(element) => { this.textElement = element; }} className={cx('hover-text')}>{text}</h3>
          </div>
        </button>
      </div>
    );
  }
}

HoverTarget.propTypes = propTypes;
HoverTarget.defaultProps = defaultProps;

export default HoverTarget;
