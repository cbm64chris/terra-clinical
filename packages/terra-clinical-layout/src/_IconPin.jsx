/* eslint-disable */
import React from 'react';

const IconPin = (props) => {
  return (
    <svg {...props} viewBox="0 0 48 48">
      <path d="M40,26.2c-1.7-1.7-5-2.6-5-2.6V0H13v23.6c0,0-3.1,0.7-5,2.6C6.2,28.1,8,31,8,31h14c0,0,0,11.8,0,13.5s0.3,3.5,2,3.5 c1.7,0,2-1.9,2-3.5S26,31,26,31h14C40,31,41.7,27.9,40,26.2z M22,23h-4V4h4V23z"/>
    </svg>
  );
};

export default IconPin;
