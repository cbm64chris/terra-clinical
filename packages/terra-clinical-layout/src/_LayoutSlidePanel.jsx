import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import 'terra-base/lib/baseStyles';
import Overlay from 'terra-overlay';

import HoverTarget from './_HoverTarget';
import styles from './LayoutSlidePanel.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * The component to display in the main content area.
   */
  background: PropTypes.node,
  /**
   * Whether or not the panel transitions should be animated.
   */
  isAnimated: PropTypes.bool,
  /**
   * Whether or not the panel should be displayed.
   */
  isOpen: PropTypes.bool,
  /**
   * Whether or not the panel transitions should be animated.
   */
  isToggleEnabled: PropTypes.bool,
    /**
   * The component to display in the main content area.
   */
  mainContent: PropTypes.node,
  /**
   * Callback when the overlay close functionality is triggered.
   */
  onRequestClose: PropTypes.func,
  /**
   * The style of panel presentation. One of `overlay`, `squish`.
   */
  panelBehavior: PropTypes.oneOf(['overlay', 'squish']),
  /**
   * The component to display in the panel content area.
   */
  panelContent: PropTypes.node,
  /**
   * Current breakpoint size.
   */
  size: PropTypes.string.isRequired,
  /**
   * Current breakpoint size.
   */
  toggleMenu: PropTypes.func,
};

const defaultProps = {
  isAnimated: false,
  isOpen: false,
  isToggleEnabled: false,
  panelBehavior: 'overlay',
};

const LayoutSlidePanel = ({
  background,
  isAnimated,
  isOpen,
  isToggleEnabled,
  mainContent,
  onRequestClose,
  panelBehavior,
  panelContent,
  size,
  toggleMenu,
  ...customProps
  }) => {
  const isTiny = size === 'tiny';
  const isSmall = size === 'small';
  const compactSize = isTiny || isSmall;
  const isOverlay = compactSize ? true : panelBehavior === 'overlay';
  const isOverlayOpen = isOpen && isOverlay && isToggleEnabled;
  const overlayBackground = compactSize ? 'dark' : 'clear';

  const slidePanelClassNames = cx([
    'layout-slide-panel',
    { 'is-open': isOpen && isToggleEnabled },
    { 'use-widescreen-style': !compactSize && isToggleEnabled },
    { 'is-overlay': isOverlay },
    { 'is-squish': !isOverlay },
    customProps.className,
  ]);

  const panelClasses = cx([
    'panel',
    { 'is-tiny': isTiny },
    { 'is-small': isSmall },
    { 'is-animated': isAnimated },
  ]);

  let panel;
  if (isToggleEnabled) {
    panel = (
      <div className={panelClasses} aria-hidden={!isOpen ? 'true' : null}>
        <HoverTarget
          onClick={toggleMenu}
          isMenuEnabled={!isOpen}
          isHoverEnabled={!compactSize && isOverlay}
          onHoverOff={() => { if (isOpen) { toggleMenu(); } }}
          onHoverOn={() => { if (!isOpen) { toggleMenu(); } }}
        >
          {panelContent}
        </HoverTarget>
      </div>
    );
  }

  return (
    <div
      {...customProps}
      className={slidePanelClassNames}
    >
      {panel}
      {background}
      <div className={cx('main')}>
        <Overlay isRelativeToContainer onRequestClose={onRequestClose} isOpen={isOverlayOpen} backgroundStyle={overlayBackground} />
        {mainContent}
      </div>
    </div>
  );
};

LayoutSlidePanel.propTypes = propTypes;
LayoutSlidePanel.defaultProps = defaultProps;

export default LayoutSlidePanel;
