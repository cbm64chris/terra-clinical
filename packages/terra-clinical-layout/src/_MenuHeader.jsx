import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Button from 'terra-button';
import 'terra-base/lib/baseStyles';

import IconPin from './_IconPin';

import './MenuHeader.scss';

const propTypes = {
  text: PropTypes.string,
  togglePin: PropTypes.func,
  isPinned: PropTypes.bool,
};

const defaultProps = {
  isPinned: false,
};

const MenuHeader = ({ text, togglePin, isPinned, ...customProps }) => {
  const headerClassNames = classNames([
    'terraClinical-MenuHeader',
    customProps.className,
  ]);

  let pinButton;
  if (togglePin && !isPinned) {
    pinButton = <Button onClick={togglePin} icon={<IconPin className="terraClinical-Unlock" />} variant="link" style={{ outline: 'none' }} />;
  }
  let unpinButton;
  if (togglePin && isPinned) {
    unpinButton = <Button onClick={togglePin} icon={<IconPin />} variant="link" style={{ outline: 'none' }} />;
  }

  const toolbarEnd = (
    <div className="terraClinical-MenuHeader-end">
      {pinButton}
      {unpinButton}
    </div>
  );

  return (
    <div {...customProps} className={headerClassNames}>
      <div className="terraClinical-MenuHeader-body">
        <h3 style={{ margin: '0' }}>{text}</h3>
      </div>
      {toolbarEnd}
    </div>
  );
};

MenuHeader.propTypes = propTypes;
MenuHeader.defaultProps = defaultProps;

export default MenuHeader;
