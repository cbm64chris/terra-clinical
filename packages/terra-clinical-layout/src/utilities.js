import breakpoints from 'terra-responsive-element/lib/breakpoints.scss';

const getBreakpointSize = (queryWidth) => {
  const width = queryWidth || window.innerWidth;
  const { small, medium, large, huge } = breakpoints;

  if (width >= huge) {
    return 'huge';
  } else if (width >= large) {
    return 'large';
  } else if (width >= medium) {
    return 'medium';
  } else if (width >= small) {
    return 'small';
  }
  return 'tiny';
};

const getCustomProps = (props, validPropTypes) => (
  Object.keys(props)
    .filter(key => !Object.keys(validPropTypes).includes(key))
    .reduce((customProps, key) => {
      customProps[key] = props[key]; // eslint-disable-line no-param-reassign
      return customProps;
    }, {})
);

export {
  getBreakpointSize,
  getCustomProps,
};
