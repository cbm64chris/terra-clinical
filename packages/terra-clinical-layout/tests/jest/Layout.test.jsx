import React from 'react';

import Layout from '../../src/Layout';
import { getBreakpointSize, getCustomProps } from '../../src/utilities';

jest.mock('../../src/utilities');

describe('Layout', () => {
  it('should render a Layout without optional props', () => {
    getBreakpointSize.mockImplementation(() => 'tiny');

    const result = shallow((
      <Layout />
    ));
    expect(result).toMatchSnapshot();
  });

  it('should render a Layout with custom props', () => {
    getBreakpointSize.mockImplementation(() => 'tiny');
    getCustomProps.mockReturnValueOnce({ 'data-test-attr': 'test-val' });

    const result = shallow((
      <Layout />
    ));
    expect(result).toMatchSnapshot();
  });

  describe('Menu Disabled', () => {
    const layout = (
      <Layout
        header={<div>Test Header</div>}
      >
        <div>Test Content</div>
      </Layout>
    );

    it('should render a Layout when tiny', () => {
      getBreakpointSize.mockImplementation(() => 'tiny');

      const result = shallow(layout);
      expect(result).toMatchSnapshot();
    });

    it('should render a Layout when small', () => {
      getBreakpointSize.mockImplementation(() => 'small');

      const result = shallow(layout);
      expect(result).toMatchSnapshot();
    });

    it('should render a Layout when medium', () => {
      getBreakpointSize.mockImplementation(() => 'medium');

      const result = shallow(layout);
      expect(result).toMatchSnapshot();
    });

    it('should render a Layout when large', () => {
      getBreakpointSize.mockImplementation(() => 'large');

      const result = shallow(layout);
      expect(result).toMatchSnapshot();
    });

    it('should render a Layout when huge', () => {
      getBreakpointSize.mockImplementation(() => 'huge');

      const result = shallow(layout);
      expect(result).toMatchSnapshot();
    });
  });

  describe('Menu Enabled', () => {
    const layout = (
      <Layout
        header={<div>Test Header</div>}
        menu={<div>Test Menu</div>}
      >
        <div>Test Content</div>
      </Layout>
    );

    it('should render a Layout when tiny', () => {
      getBreakpointSize.mockImplementation(() => 'tiny');

      const result = shallow(layout);
      expect(result).toMatchSnapshot();
    });

    it('should render a Layout when small', () => {
      getBreakpointSize.mockImplementation(() => 'small');

      const result = shallow(layout);
      expect(result).toMatchSnapshot();
    });

    it('should render a Layout when medium', () => {
      getBreakpointSize.mockImplementation(() => 'medium');

      const result = shallow(layout);
      expect(result).toMatchSnapshot();
    });

    it('should render a Layout when large', () => {
      getBreakpointSize.mockImplementation(() => 'large');

      const result = shallow(layout);
      expect(result).toMatchSnapshot();
    });

    it('should render a Layout when huge', () => {
      getBreakpointSize.mockImplementation(() => 'huge');

      const result = shallow(layout);
      expect(result).toMatchSnapshot();
    });
  });
});
