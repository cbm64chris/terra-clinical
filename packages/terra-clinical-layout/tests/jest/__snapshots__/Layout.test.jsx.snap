// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Menu Disabled should render a Layout when huge 1`] = `
<ContentContainer
  fill={true}
  header={
    <div
      layoutConfig={
        Object {
          "isCompactLayout": false,
          "menuIsOpen": false,
          "size": "huge",
          "toggleMenu": false,
        }
      }
    >
      Test Header
    </div>
  }
>
  <MenuSlidePanel
    isAnimated={true}
    isOpen={false}
    isToggleEnabled={false}
    mainContent={
      <ContentContainer
        fill={true}
        header={false}
      >
        <div
          layoutConfig={
            Object {
              "isCompactLayout": false,
              "menuIsOpen": false,
              "size": "huge",
              "toggleMenu": [Function],
            }
          }
        >
          Test Content
        </div>
      </ContentContainer>
    }
    onRequestClose={[Function]}
    panelBehavior="overlay"
    panelContent={
      <ContentContainer
        fill={true}
        header={
          <MenuHeader
            isPinned={false}
            togglePin={[Function]}
          />
        }
      />
    }
    size="huge"
    toggleMenu={[Function]}
  />
</ContentContainer>
`;

exports[`Layout Menu Disabled should render a Layout when large 1`] = `
<ContentContainer
  fill={true}
  header={
    <div
      layoutConfig={
        Object {
          "isCompactLayout": false,
          "menuIsOpen": false,
          "size": "large",
          "toggleMenu": false,
        }
      }
    >
      Test Header
    </div>
  }
>
  <MenuSlidePanel
    isAnimated={true}
    isOpen={false}
    isToggleEnabled={false}
    mainContent={
      <ContentContainer
        fill={true}
        header={false}
      >
        <div
          layoutConfig={
            Object {
              "isCompactLayout": false,
              "menuIsOpen": false,
              "size": "large",
              "toggleMenu": [Function],
            }
          }
        >
          Test Content
        </div>
      </ContentContainer>
    }
    onRequestClose={[Function]}
    panelBehavior="overlay"
    panelContent={
      <ContentContainer
        fill={true}
        header={
          <MenuHeader
            isPinned={false}
            togglePin={[Function]}
          />
        }
      />
    }
    size="large"
    toggleMenu={[Function]}
  />
</ContentContainer>
`;

exports[`Layout Menu Disabled should render a Layout when medium 1`] = `
<ContentContainer
  fill={true}
  header={
    <div
      layoutConfig={
        Object {
          "isCompactLayout": false,
          "menuIsOpen": false,
          "size": "medium",
          "toggleMenu": false,
        }
      }
    >
      Test Header
    </div>
  }
>
  <MenuSlidePanel
    isAnimated={true}
    isOpen={false}
    isToggleEnabled={false}
    mainContent={
      <ContentContainer
        fill={true}
        header={false}
      >
        <div
          layoutConfig={
            Object {
              "isCompactLayout": false,
              "menuIsOpen": false,
              "size": "medium",
              "toggleMenu": [Function],
            }
          }
        >
          Test Content
        </div>
      </ContentContainer>
    }
    onRequestClose={[Function]}
    panelBehavior="overlay"
    panelContent={
      <ContentContainer
        fill={true}
        header={
          <MenuHeader
            isPinned={false}
            togglePin={[Function]}
          />
        }
      />
    }
    size="medium"
    toggleMenu={[Function]}
  />
</ContentContainer>
`;

exports[`Layout Menu Disabled should render a Layout when small 1`] = `
<ContentContainer
  fill={true}
  header={false}
>
  <MenuSlidePanel
    isAnimated={true}
    isOpen={false}
    isToggleEnabled={false}
    mainContent={
      <ContentContainer
        fill={true}
        header={
          <div
            layoutConfig={
              Object {
                "isCompactLayout": true,
                "menuIsOpen": false,
                "size": "small",
                "toggleMenu": false,
              }
            }
          >
            Test Header
          </div>
        }
      >
        <div
          layoutConfig={
            Object {
              "isCompactLayout": true,
              "menuIsOpen": false,
              "size": "small",
              "toggleMenu": [Function],
            }
          }
        >
          Test Content
        </div>
      </ContentContainer>
    }
    onRequestClose={[Function]}
    panelBehavior="overlay"
    panelContent={
      <ContentContainer
        fill={true}
      />
    }
    size="small"
    toggleMenu={[Function]}
  />
</ContentContainer>
`;

exports[`Layout Menu Disabled should render a Layout when tiny 1`] = `
<ContentContainer
  fill={true}
  header={false}
>
  <MenuSlidePanel
    isAnimated={true}
    isOpen={false}
    isToggleEnabled={false}
    mainContent={
      <ContentContainer
        fill={true}
        header={
          <div
            layoutConfig={
              Object {
                "isCompactLayout": true,
                "menuIsOpen": false,
                "size": "tiny",
                "toggleMenu": false,
              }
            }
          >
            Test Header
          </div>
        }
      >
        <div
          layoutConfig={
            Object {
              "isCompactLayout": true,
              "menuIsOpen": false,
              "size": "tiny",
              "toggleMenu": [Function],
            }
          }
        >
          Test Content
        </div>
      </ContentContainer>
    }
    onRequestClose={[Function]}
    panelBehavior="overlay"
    panelContent={
      <ContentContainer
        fill={true}
      />
    }
    size="tiny"
    toggleMenu={[Function]}
  />
</ContentContainer>
`;

exports[`Layout Menu Enabled should render a Layout when huge 1`] = `
<ContentContainer
  fill={true}
  header={
    <div
      layoutConfig={
        Object {
          "isCompactLayout": false,
          "menuIsOpen": false,
          "size": "huge",
          "toggleMenu": false,
        }
      }
    >
      Test Header
    </div>
  }
>
  <MenuSlidePanel
    isAnimated={true}
    isOpen={false}
    isToggleEnabled={true}
    mainContent={
      <ContentContainer
        fill={true}
        header={false}
      >
        <div
          layoutConfig={
            Object {
              "isCompactLayout": false,
              "menuIsOpen": false,
              "size": "huge",
              "toggleMenu": [Function],
            }
          }
        >
          Test Content
        </div>
      </ContentContainer>
    }
    onRequestClose={[Function]}
    panelBehavior="overlay"
    panelContent={
      <ContentContainer
        fill={true}
        header={
          <MenuHeader
            isPinned={false}
            togglePin={[Function]}
          />
        }
      >
        <div
          layoutConfig={
            Object {
              "isCompactLayout": false,
              "menuIsOpen": false,
              "size": "huge",
              "toggleMenu": [Function],
            }
          }
        >
          Test Menu
        </div>
      </ContentContainer>
    }
    size="huge"
    toggleMenu={[Function]}
  />
</ContentContainer>
`;

exports[`Layout Menu Enabled should render a Layout when large 1`] = `
<ContentContainer
  fill={true}
  header={
    <div
      layoutConfig={
        Object {
          "isCompactLayout": false,
          "menuIsOpen": false,
          "size": "large",
          "toggleMenu": false,
        }
      }
    >
      Test Header
    </div>
  }
>
  <MenuSlidePanel
    isAnimated={true}
    isOpen={false}
    isToggleEnabled={true}
    mainContent={
      <ContentContainer
        fill={true}
        header={false}
      >
        <div
          layoutConfig={
            Object {
              "isCompactLayout": false,
              "menuIsOpen": false,
              "size": "large",
              "toggleMenu": [Function],
            }
          }
        >
          Test Content
        </div>
      </ContentContainer>
    }
    onRequestClose={[Function]}
    panelBehavior="overlay"
    panelContent={
      <ContentContainer
        fill={true}
        header={
          <MenuHeader
            isPinned={false}
            togglePin={[Function]}
          />
        }
      >
        <div
          layoutConfig={
            Object {
              "isCompactLayout": false,
              "menuIsOpen": false,
              "size": "large",
              "toggleMenu": [Function],
            }
          }
        >
          Test Menu
        </div>
      </ContentContainer>
    }
    size="large"
    toggleMenu={[Function]}
  />
</ContentContainer>
`;

exports[`Layout Menu Enabled should render a Layout when medium 1`] = `
<ContentContainer
  fill={true}
  header={
    <div
      layoutConfig={
        Object {
          "isCompactLayout": false,
          "menuIsOpen": false,
          "size": "medium",
          "toggleMenu": false,
        }
      }
    >
      Test Header
    </div>
  }
>
  <MenuSlidePanel
    isAnimated={true}
    isOpen={false}
    isToggleEnabled={true}
    mainContent={
      <ContentContainer
        fill={true}
        header={false}
      >
        <div
          layoutConfig={
            Object {
              "isCompactLayout": false,
              "menuIsOpen": false,
              "size": "medium",
              "toggleMenu": [Function],
            }
          }
        >
          Test Content
        </div>
      </ContentContainer>
    }
    onRequestClose={[Function]}
    panelBehavior="overlay"
    panelContent={
      <ContentContainer
        fill={true}
        header={
          <MenuHeader
            isPinned={false}
            togglePin={[Function]}
          />
        }
      >
        <div
          layoutConfig={
            Object {
              "isCompactLayout": false,
              "menuIsOpen": false,
              "size": "medium",
              "toggleMenu": [Function],
            }
          }
        >
          Test Menu
        </div>
      </ContentContainer>
    }
    size="medium"
    toggleMenu={[Function]}
  />
</ContentContainer>
`;

exports[`Layout Menu Enabled should render a Layout when small 1`] = `
<ContentContainer
  fill={true}
  header={false}
>
  <MenuSlidePanel
    isAnimated={true}
    isOpen={false}
    isToggleEnabled={true}
    mainContent={
      <ContentContainer
        fill={true}
        header={
          <div
            layoutConfig={
              Object {
                "isCompactLayout": true,
                "menuIsOpen": false,
                "size": "small",
                "toggleMenu": [Function],
              }
            }
          >
            Test Header
          </div>
        }
      >
        <div
          layoutConfig={
            Object {
              "isCompactLayout": true,
              "menuIsOpen": false,
              "size": "small",
              "toggleMenu": [Function],
            }
          }
        >
          Test Content
        </div>
      </ContentContainer>
    }
    onRequestClose={[Function]}
    panelBehavior="overlay"
    panelContent={
      <ContentContainer
        fill={true}
      >
        <div
          layoutConfig={
            Object {
              "isCompactLayout": true,
              "menuIsOpen": false,
              "size": "small",
              "toggleMenu": [Function],
            }
          }
        >
          Test Menu
        </div>
      </ContentContainer>
    }
    size="small"
    toggleMenu={[Function]}
  />
</ContentContainer>
`;

exports[`Layout Menu Enabled should render a Layout when tiny 1`] = `
<ContentContainer
  fill={true}
  header={false}
>
  <MenuSlidePanel
    isAnimated={true}
    isOpen={false}
    isToggleEnabled={true}
    mainContent={
      <ContentContainer
        fill={true}
        header={
          <div
            layoutConfig={
              Object {
                "isCompactLayout": true,
                "menuIsOpen": false,
                "size": "tiny",
                "toggleMenu": [Function],
              }
            }
          >
            Test Header
          </div>
        }
      >
        <div
          layoutConfig={
            Object {
              "isCompactLayout": true,
              "menuIsOpen": false,
              "size": "tiny",
              "toggleMenu": [Function],
            }
          }
        >
          Test Content
        </div>
      </ContentContainer>
    }
    onRequestClose={[Function]}
    panelBehavior="overlay"
    panelContent={
      <ContentContainer
        fill={true}
      >
        <div
          layoutConfig={
            Object {
              "isCompactLayout": true,
              "menuIsOpen": false,
              "size": "tiny",
              "toggleMenu": [Function],
            }
          }
        >
          Test Menu
        </div>
      </ContentContainer>
    }
    size="tiny"
    toggleMenu={[Function]}
  />
</ContentContainer>
`;

exports[`Layout should render a Layout with custom props 1`] = `
<ContentContainer
  data-test-attr="test-val"
  fill={true}
  header={false}
>
  <MenuSlidePanel
    isAnimated={true}
    isOpen={false}
    isToggleEnabled={false}
    mainContent={null}
    onRequestClose={[Function]}
    panelBehavior="overlay"
    panelContent={
      <ContentContainer
        fill={true}
      />
    }
    size="tiny"
    toggleMenu={[Function]}
  />
</ContentContainer>
`;

exports[`Layout should render a Layout without optional props 1`] = `
<ContentContainer
  fill={true}
  header={false}
>
  <MenuSlidePanel
    isAnimated={true}
    isOpen={false}
    isToggleEnabled={false}
    mainContent={null}
    onRequestClose={[Function]}
    panelBehavior="overlay"
    panelContent={
      <ContentContainer
        fill={true}
      />
    }
    size="tiny"
    toggleMenu={[Function]}
  />
</ContentContainer>
`;
